core/src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/custom_block/noteblock/RegularNoteBlock.java

```diff
@@ -35,7 +36,19 @@ public TuneBlock(Block block, @Nullable Player player) {
         this.isPowered = container.getOrDefault(poweredKey, PersistentDataType.BOOLEAN, false);
         this.note = container.getOrDefault(noteKey, PersistentDataType.BYTE, (byte) 0);
         this.pitch = (float) Math.pow(2.0F, (note - 12F) / 12F);
-        this.sound = NMSHandlers.getHandler().getNoteBlockInstrument(block);
+        this.sound = switch (blockAbove.getType()) {
+            case SKELETON_SKULL -> "block.note_block.imitate.skeleton";
+            case PIGLIN_HEAD -> "block.note_block.imitate.piglin";
+            case ZOMBIE_HEAD -> "block.note_block.imitate.zombie";
+            case CREEPER_HEAD -> "block.note_block.imitate.creeper";
+            case DRAGON_HEAD -> "block.note_block.imitate.ender_dragon";
+            case WITHER_SKELETON_SKULL -> "block.note_block.imitate.wither_skeleton";
```

Comment:
player head should alsdo be taken into account and read from the data-component on all supported versions